identity/authentication.go

```diff
@@ -151,12 +152,12 @@ func unmarshalIdentityResponse(resp *http.Response) (identityResp *common.Identi
 		return
 	}
 
-	defer func() {
-		if errClose := resp.Body.Close(); errClose != nil {
-			log.ErrorC("error closing response body", errClose, nil)
-		}
-	}()
-
 	err = json.Unmarshal(b, &identityResp)
 	return
 }
+
+func CloseResponse(ctx context.Context, resp *http.Response, data log.Data) {
```

Comment:
this is exported, consider:
```suggestion
func closeResponse(ctx context.Context, resp *http.Response, data log.Data) {
```